course/include/isometry/isometry.hpp

```diff
@@ -9,16 +9,69 @@
 #include <cmath>
 #include <initializer_list>
 #include <iostream>
+#include <isometry/matrix3.hpp>
 #include <sstream>
 #include <string>
+#include <vector>
 
 namespace ekumen {
 
 namespace math {
 
-/*
- * your code here
- */
+class Isometry {
+ public:
+  /// Constructs an Isometry with default values.
+  Isometry() : rotation_matrix{Matrix3::kIdentity} {};
+
+  /// Constructs an Isometry.
+  /// @param r_vector Initial value of the translation vector.
+  /// @param r_matrix Initial value of the rotation matrix.
+  Isometry(const Vector3& r_vector, const Matrix3& r_matrix)
+      : translation_vector{r_vector}, rotation_matrix{r_matrix}{};
+
+  /// Gets the Isometry matrix from a vector
+  /// @param r_vector Value of the translation vector.
+  static Isometry fromTranslation(const Vector3& r_vector);
+
+  /// Gets the Isometry matrix from Euler angles
+  /// @param roll Value of the roll angle.
+  /// @param pitch Value of the pitch angle.
+  /// @param yaw Value of the yaw angle.
+  static Isometry fromEulerAngles(const double& roll, const double& pitch,
+                                  const double& yaw);
+
+  Vector3 translation() const;
+
+  Vector3 transform(const Vector3& r_vector) const;
+
+  Isometry inverse() const;
+
+  Matrix3 rotation() const;
+
+  /// Gets the Isometry matrix from rotation a certain angle around a given direction unitary-vector
+  /// @param r_vector Unitary Vector to rotate around.
+  /// @param angle Value of the angle of rotation.
+  static Isometry rotateAround(const Vector3& r_vector, const double& angle);
```

Comment:
nit: passing `double` by reference is on the limit of what's needed. They are still small enough to be passed by copy. Don't change it, though. It's ok.